scripts/pylib/twister/twisterlib.py

```diff
@@ -3365,40 +3398,50 @@ def load_quarantine(self, file):
-                    if row["status"] in filter_status:
-                        continue
-                    test = row["test"]
+    def load_from_file(self, file, filter_platform=[]):
+        with open(file, "r") as json_test_plan:
+            jtp = json.load(json_test_plan)
+            instance_list = []
+            for ts in jtp.get("testsuites", []):
+                logger.debug(f"loading {ts['name']}...")
+                testsuite = ts["name"]
 
-                    platform = self.get_platform(row["platform"])
-                    if filter_platform and platform.name not in filter_platform:
-                        continue
-                    instance = TestInstance(self.testcases[test], platform, self.outdir)
-                    if self.device_testing:
-                        tfilter = 'runnable'
-                    else:
-                        tfilter = 'buildable'
-                    instance.run = instance.check_runnable(
-                        self.enable_slow,
-                        tfilter,
-                        self.fixtures
-                    )
-                    instance.create_overlay(platform, self.enable_asan, self.enable_ubsan, self.enable_coverage, self.coverage_platform)
-                    instance_list.append(instance)
-                self.add_instances(instance_list)
+                platform = self.get_platform(ts["platform"])
+                if filter_platform and platform.name not in filter_platform:
+                    continue
+                instance = TestInstance(self.testsuites[testsuite], platform, self.outdir)
+                if self.device_testing:
+                    tfilter = 'runnable'
+                else:
+                    tfilter = 'buildable'
+                instance.run = instance.check_runnable(
+                    self.enable_slow,
+                    tfilter,
+                    self.fixtures
+                )
 
-        except KeyError as e:
-            logger.error("Key error while parsing tests file.({})".format(str(e)))
-            sys.exit(2)
+                instance.metrics['handler_time'] = ts.get('execution_time', 0)
+                instance.metrics['ram_size'] = ts.get("ram_size", 0)
+                instance.metrics['rom_size']  = ts.get("rom_size",0)
 
-        except FileNotFoundError as e:
-            logger.error("Couldn't find input file with list of tests. ({})".format(e))
-            sys.exit(2)
+                status = ts.get('status', None)
+                reason = ts.get("reason", "Unknown")
+                if status in ["error", "failed"]:
+                    instance.status = None
+                else:
+                    instance.status = status
+                instance.reason = reason
+
+                for tc in ts.get('testcases', []):
+                    identifier = tc['identifier']
+                    tc_status = tc.get('status', None)
+                    tc_reason = tc.get('reason')
```

Comment:
we should save each testcase's execution_time too, otherwise, when run twister with --only-failed option, all testcases' execution time will be updated to default 0.